app/models/mixins/retirement_mixin.rb

```diff
@@ -229,6 +238,10 @@ def error_retiring?
 
   private
 
+  def find_requester(requester_context, q_options = {})
+    requester_context == "system" ? q_options[:user_id] : User.current_user
```

Comment:
@d-m-u  This could still be nil if the q_options[:userid] is not set

Reply:
Yeah, @tinaafitz, I'm not sure how to handle that

Reply:
@mkanoor @d-m-u How about we remove q_options from the find_requester method, and instead use a ```system_requester``` method to to return the objects owner user object?